### PR TITLE
Add CustomDigestItem type

### PR DESCRIPTION
**Pull request by vgeddes**
The outbound queue pallet inserts a merkle root as a custom digest item into the parachain header.

To play nice with other bridges, this type allows these bridges to insert their own custom digest items, while making sure they don't conflict with each other.

### CODE REVIEW

**Review by alistair-singh — APPROVED** [review]

**Comment by yrong** [comment]
Comment in
https://github.com/Snowfork/polkadot-sdk/pull/65/files#r1427503626

File: contracts/src/Verification.sol [hunk 1]
```diff
@@ -133,8 +136,17 @@ library Verification {
     {
         for (uint256 i = 0; i < header.digestItems.length; i++) {
             DigestItem memory item = header.digestItems[i];
-            if (item.kind == DIGEST_ITEM_OTHER && item.data.length == 32 && commitment == bytes32(item.data)) {
-                return true;
+            if (item.kind == DIGEST_ITEM_OTHER && item.data.length == 33 && item.data[0] == DIGEST_ITEM_PREFIX) {
+                bytes memory commitment_in_header = new bytes(32);
+                for (uint256 j = 1; j < 33;) {
```
Comment (vgeddes) [inline contracts/src/Verification.sol:141]: Can't we use array slices here? https://docs.soliditylang.org/en/v0.6.0/types.html#array-slices

Comment (yrong) [reply contracts/src/Verification.sol:141]: > As of now, array slices are only implemented for calldata arrays. 

As noted on the above page, or error is raised as follows:
`Error (1227): Index range access is only supported for dynamic calldata arrays.`

Comment (vgeddes) [reply contracts/src/Verification.sol:141]: Right, so isn't `header.digestItems[i].data` exactly that?

Comment (vgeddes) [reply contracts/src/Verification.sol:141]: If we remove this line, and reference the calldata directly, then it should compile?
```
 DigestItem memory item = header.digestItems[i];
```

Comment (yrong) [reply contracts/src/Verification.sol]: https://github.com/Snowfork/snowbridge/pull/1053/commits/7ba6c0ec0a3cd78d038578db1849bf9f604ab06e

**Review by yrong — APPROVED** [review]
All smoke tests work as expected, should be fine now.

**Review by claravanstaden — APPROVED** [review]
Cool!

File: contracts/src/Verification.sol [hunk 2]
```diff
@@ -76,6 +76,9 @@ library Verification {
     uint256 public constant DIGEST_ITEM_PRERUNTIME = 6;
     uint256 public constant DIGEST_ITEM_RUNTIME_ENVIRONMENT_UPDATED = 8;
 
+    /// @dev Prefix of enum variants of DigestItem(0 is reserved for snowbridge)
+    bytes1 public constant DIGEST_ITEM_PREFIX = 0;
```
Comment (vgeddes) [inline contracts/src/Verification.sol:80]: Prefer this naming:
```suggestion
    bytes1 public constant DIGEST_ITEM_OTHER_SNOWBRIDGE = 0x00;
```

File: web/packages/test/scripts/configure-assethub.sh
Comment (vgeddes) [inline web/packages/test/scripts/configure-assethub.sh:1]: hmm, this file does not seem relevant to this PR?

File: contracts/src/Verification.sol (hunk 2 above)
Comment (yrong) [reply contracts/src/Verification.sol]: [d0bb6c9](https://github.com/Snowfork/snowbridge/pull/1053/commits/d0bb6c91d135c6be705ce85f43e166d29b6960a0)

File: web/packages/test/scripts/configure-assethub.sh
Comment (yrong) [reply web/packages/test/scripts/configure-assethub.sh:1]: It's required for `reserve_transfer` but is already covered in another PR, so removed here.
